google-ads-examples/src/main/java/com/google/ads/googleads/examples/accountmanagement/GetAccountHierarchy.java

```diff
@@ -0,0 +1,260 @@
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the examples.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param managerId managerCustomerId the customer ID.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, Long managerId) {
+    List<Long> seedCustomerIds = new ArrayList<>();
+    if (!(managerId == null)) {
+      seedCustomerIds.add(managerId);
+    } else {
+      // Issues a request for listing all accessible customers by this authenticated Google account.
+      try (CustomerServiceClient customerServiceClient =
+          googleAdsClient.getLatestVersion().createCustomerServiceClient()) {
+        ListAccessibleCustomersResponse accessibleCustomers =
+            customerServiceClient.listAccessibleCustomers(
+                ListAccessibleCustomersRequest.newBuilder().build());
+        System.out.println(
+            "No manager customer ID is specified. The example will print the "
+                + "hierarchies of all accessible customer IDs:");
+
+        for (String customerResourceName : accessibleCustomers.getResourceNamesList()) {
+          long customer =
+              customerServiceClient.getCustomer(customerResourceName).getId().getValue();
+          System.out.println(customer);
+          seedCustomerIds.add(customer);
+        }
+      }
+    }
+
+    List<Long> unprocessedCustomerIds = new ArrayList<>();
+    for (long seedCustomerId : seedCustomerIds) {
+      // Instantiates several variables used later.
+      Map<Long, List<CustomerClient>> customerIdsToChildAccounts = new HashMap<>();
+      CustomerClient rootCustomerClient = null;
+      int depth = 0;
+      // Adds the seed customer ID to the list of IDs to be processed.
+      unprocessedCustomerIds.add(seedCustomerId);
+
+      // Creates the Google Ads Service client.
+      try (GoogleAdsServiceClient googleAdsServiceClient =
+          googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+        // Creates a query that retrieves all child accounts of the manager specified in search
+        // calls below.
+        String query =
+            "SELECT customer_client.client_customer, customer_client.level, "
+                + "customer_client.manager, customer_client.descriptive_name, "
+                + "customer_client.currency_code, customer_client.time_zone, "
+                + "customer_client.id "
+                + "FROM customer_client "
+                + "WHERE customer_client.level <= 1";
+
+        // Performs a breadth-first search algorithm to build a mapping of managers to their
+        // child accounts. The map keys are manager IDs, and the values are lists of those
+        // managers' direct child accounts, as represented by CustomerClient objects.
+        while (!unprocessedCustomerIds.isEmpty()) {
+          long customerId = unprocessedCustomerIds.get(0);
+          unprocessedCustomerIds.remove(0);
```

Comment:
Can do this in one line. `List.remove(int index)` will return the removed object.

Reply:
unprocessedCustomerIds (now named managerAccountsToSearch) has been changed to a set, so this has changed and it now a moot point.